### PR TITLE
spring rest hw

### PR DESCRIPTION
**Pull request by etezaadi**
Configured Spring MVC and added all dependencies;
Created all three controllers (MovieController, CinemaHallController, MovieSessionController) and implemented their methods;
Created DTOs models and mappers for their.

### CODE REVIEW

File: src/main/java/mate/academy/spring/config/WebConfig.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.springframework.web.servlet.view.InternalResourceViewResolver;
+import org.springframework.web.servlet.view.JstlView;
+
+@EnableWebMvc
+@Configuration
+@ComponentScan(basePackages = "mate.academy.spring.controller")
+public class WebConfig implements WebMvcConfigurer {
+
```
Comment (Nick97-git) [inline src/main/java/mate/academy/spring/config/WebConfig.java:15]: remove empty line

File: src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -57,4 +58,48 @@ public Optional<MovieSession> get(Long id) {
...
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.delete(movieSession);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't delete a movie session by id: "
```
Comment (Nick97-git) [inline src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java:75]:
```suggestion
            throw new DataProcessingException("Can't delete a movie session: "
```

File: src/main/java/mate/academy/spring/service/mapper/MovieDtoMapper.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package mate.academy.spring.service.mapper;
+
+import mate.academy.spring.model.Movie;
+import mate.academy.spring.model.dto.MovieRequestDto;
+import mate.academy.spring.model.dto.MovieResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieDtoMapper {
+
```
Comment (Nick97-git) [inline src/main/java/mate/academy/spring/service/mapper/MovieDtoMapper.java:10]: remove empty line

**Review by Nick97-git — CHANGES_REQUESTED** [review]

**Comment by etezaadi** [comment]
@Nick97-git already fixed.

**Review by Nick97-git — APPROVED** [review]

**Pull request closed without merging**